test/source/mock/lib/api.ts

```diff
@@ -225,12 +225,17 @@ export class Api<REQ, RES> {
   };
 
   private throttledResponse = async (response: http.ServerResponse, data: Buffer) => {
-    const chunkSize = 100 * 1024;
-    for (let i = 0; i < data.length; i += chunkSize) {
-      const chunk = data.slice(i, i + chunkSize);
-      response.write(chunk);
-      if (i > 0) {
-        await Util.sleep(this.throttleChunkMsDownload / 1000);
+    // If google oauth2 login, then redirect to url
+    if (/^https:\/\/www\.google.com\/robots\.txt/.test(data.toString())) {
+      response.writeHead(302, { 'Location': data.toString() });
+    } else {
+      const chunkSize = 100 * 1024;
+      for (let i = 0; i < data.length; i += chunkSize) {
+        const chunk = data.slice(i, i + chunkSize);
+        response.write(chunk);
+        if (i > 0) {
+          await Util.sleep(this.throttleChunkMsDownload / 1000);
+        }
```

Comment:
We can probably come up with a cleaner solution here. I suppose, in mocked API, we shouldn't need to deal with the `robots.txt` location at all. We already have a different destination for oauth2 dialog in mock, and we can also have a different place instead of google/robots.txt for the redirect, something that maybe wouldn't need this code here.

However, I haven't fully examined how this is used, therefore I'm not 100% sure if my suggestion is appropriate.

Reply:
Yes, looks like `https://www.google.com/robots.txt` is actually accessed (queried) in mock tests which we try to keep isolated.
This is not necessary as we have the code to replace URLs in mock manifests.
We're replacing `https://mail.google.com` with `https://gmail.localhost:8001` in https://github.com/FlowCrypt/flowcrypt-browser/blob/59bc0b3890294f05bd41f1bcf0eef6dbd25294ca/tooling/build-types-and-manifests.ts#L128
We can also replace `https://www.google.com` with `https://google.localhost:8001` (different host but still resolved to 127.0.0.1)